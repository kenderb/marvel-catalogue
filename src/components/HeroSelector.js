import React from 'react';
import PropTypes from 'prop-types';

const HeroSelector = ({
  image, heroName, onClickHandler, selected,
}) => (
  <button
    type="button"
    onClick={() => onClickHandler(heroName)}
    style={{ backgroundColor: selected === heroName ? 'red' : 'transparent' }}
  >
    <img src={image} alt={heroName} />
  </button>
);

HeroSelector.propTypes = {
  image: PropTypes.string.isRequired,
  heroName: PropTypes.string.isRequired,
  selected: PropTypes.string.isRequired,
  onClickHandler: PropTypes.func.isRequired,
};

export default HeroSelector;
