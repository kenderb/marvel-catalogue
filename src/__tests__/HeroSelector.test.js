import { create } from 'react-test-renderer';
import HeroSelector from '../components/HeroSelector';

test('HeroSelector renders correctly', () => {
  const tree = create(<HeroSelector />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('HeroSelector is not a string', () => {
  const tree = create(<HeroSelector />).toJSON();
  expect(tree).not.toBe(typeof String);
});
