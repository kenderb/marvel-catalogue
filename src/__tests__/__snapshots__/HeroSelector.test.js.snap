// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSelector renders correctly 1`] = `
<button
  className="sc-hKgILt xorqq"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#f7ba43",
    }
  }
  type="button"
>
  <img
    alt=""
    src=""
  />
</button>
`;
